selfdrive/car/toyota/carcontroller.py

```diff
@@ -10,33 +10,33 @@
 VisualAlert = car.CarControl.HUDControl.VisualAlert
 
 
-def accel_hysteresis(accel, accel_steady, enabled):
-
-  # for small accel oscillations within ACCEL_HYST_GAP, don't change the accel command
-  if not enabled:
-    # send 0 when disabled, otherwise acc faults
-    accel_steady = 0.
-  elif accel > accel_steady + CarControllerParams.ACCEL_HYST_GAP:
-    accel_steady = accel - CarControllerParams.ACCEL_HYST_GAP
-  elif accel < accel_steady - CarControllerParams.ACCEL_HYST_GAP:
-    accel_steady = accel + CarControllerParams.ACCEL_HYST_GAP
-  accel = accel_steady
+def coast_accel(speed):  # given a speed, output coasting acceleration
+  points = [[0.01, 0.3], [.21, .425], [.3107, .535], [.431, .555],
+            [.777, .438], [1.928, 0.265], [2.66, -0.179],
+            [3.336, -0.250], [MIN_ACC_SPEED, -0.145]]
+  return interp(speed, *zip(*points))
 
-  return accel, accel_steady
 
+def compute_gb_pedal(accel, speed, braking):
+  def accel_to_gas(a_ego, v_ego):
+    _a3, _a4, _a5, _a6, _a7, _a8, _s1, _s2, _s3, _s4, _offset = [-0.0034109221270790142, -0.02989942810035373, 0.002005326552420498, 0.1356381902353583, 0.0014222019070588158, 0.008436894892099946,
+                                                                 0.0009439048033890968, -0.0017786568461504919, 0.002986433642380856, 0.021810785976030644, -0.007020501995388009]
+    speed_part = (_s1 * a_ego + _s2) * v_ego ** 2 + (_s3 * a_ego + _s4) * v_ego
+    accel_part = (_a7 * v_ego + _a8) * a_ego ** 4 + (_a3 * v_ego + _a4) * a_ego ** 3 + _a5 * a_ego ** 2 + _a6 * a_ego
+    ret = speed_part + accel_part + _offset
+    return ret
 
-def coast_accel(speed):  # given a speed, output coasting acceleration
-  points = [[0.0, 0.03], [.166, .424], [.335, .568],
-            [.731, .440], [1.886, 0.262], [2.809, -0.207],
-            [3.443, -0.249], [MIN_ACC_SPEED, -0.145]]
-  return interp(speed, *zip(*points))
+  coast = coast_accel(speed)
+  spread = 0.2
 
+  if accel >= coast:
+    gas = accel_to_gas(accel, speed)
+    x = accel - coast  # start at 0 when at coast accel
+    if spread > x >= 0:  # ramp up gas output using s-shaped curve from coast accel to coast + spread
+      gas *= 1 / (1 + (x / (spread - x)) ** -3) if x != 0 else 0
+    return gas if not braking else gas / 2.0  # while car is braking reduce gas output
```

Comment:
With the new brakeLights message I found we know if the car is actually actuating brakes for TSS1 Corolla (not deriving from aEgo or requested accel which is nice) so we reduce output until the car releases brakes or the pedal applies enough output to move us. This reduces jerking since if you apply gas, all brakes are released immediately from the car